packages/core/src/util/hook.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+const iitmHook = require('./iitmHook');
+const requireHook = require('./requireHook');
+// eslint-disable-next-line import/no-useless-path-segments
+const isESMApp = require('./esm').isESMApp;
+
+/*
+ * RequireHook
+ *
+ * Provides support for all CommonJS (CJS) modules in CJS applications, even if the CJS version of
+ * pureEsmLibraries is used in CJS apps.
```

Comment:
> even if the CJS version of pureEsmLibraries is used in CJS apps.

Its hard to understand that pureEsmLibraries is the array below. Can you please put the array on top & add something like this "pureEsmLibraries"?

And I think I know what you are trying to say, but its not really clear. Could you please pimp?